tools/update_roles.php

```diff
@@ -0,0 +1,659 @@
+ */
+
+require_once __DIR__ . "/../vendor/autoload.php";
+require_once __DIR__ . "/generic_includes.php";
+
+$roles       = getRoles();
+$permissions = getPermissions();
+
+// User did not provide any args
+if (count($argv) === 1) {
+    syntaxIncorrect();
+    exit(1);
+}
+
+// User asking for the list of roles
+if (isset($argv[1]) && $argv[1] === "roles") {
+    if (!empty($roles)) {
+        echo "The roles in the database are:\n\n";
+        prettyPrint($roles);
+    } else {
+        echo "There are currently no roles in the database.\n";
+    }
+
+    exit(2);
+}
+
+// User asking for the list of permissions
+if (isset($argv[1]) && $argv[1] === "permissions") {
+    echo "The permissions in the database are:\n\n";
+    prettyPrint($permissions);
+    exit(3);
+}
+
+// User asking to reassign roles based on the user permissions
+if (isset($argv[1]) && $argv[1] === "refresh") {
+    refresh();
+    exit(4);
+}
+
+// Arg syntax not correct
+if (isset($argv[1]) && !isRole($argv[1])) {
+    syntaxIncorrect();
+    exit(5);
+}
+
+$role = $argv[1];
+
+// User asking for the permissions associated with a role
+if (count($argv) === 2) {
+    echo "The permissions for the $role role are:\n\n";
+    prettyPrint(getRolePermissions($role));
+    exit(6);
+}
+
+// User looking to add a permission to a role
+if (isset($argv[2]) && $argv[2] === "add") {
+    if (isset($argv[3])) {
+        $permission = $argv[3];
+        if (isPermission($permission)) {
+            addPermission($role, $permission);
+            exit(7);
+        } else {
+            echo "$permission is not a valid permission.\n";
+            exit(8);
+        }
+    } else {
+        echo "You must include the permission you would like to add.\n";
+        exit(9);
+    }
+}
+
+// User looking to remove a permission from a role
+if (isset($argv[2]) && $argv[2] === "remove") {
+    if (isset($argv[3])) {
+        $permission = $argv[3];
+        if (isPermission($permission)) {
+            removePermission($role, $permission);
+            exit();
+        } else {
+            echo "$permission is not a valid permission.\n";
+            exit(10);
+        }
+    } else {
+        echo "You must include the permission you would like to remove.\n";
+        exit(11);
+    }
+}
+
+syntaxIncorrect();
+exit(12);
+
+/**
+ * SyntaxIncorrect
+ *
+ * Prints a message to users with the argument instructions
+ *
+ * @return nothing
+ */
+function syntaxIncorrect()
+{
+    echo "You have not used the correct argument syntax for this script.
+
+To see the roles in the database:        php update_roles.php roles
+To see the permissions in the database:  php update_roles.php permissions
+To see the permissions for a role:       php update_roles.php [role]
+To add a permission to the role:         php update_roles.php [role] add [perm]
+To remove a permission from the role:    php update_roles.php [role] remove [perm]
+To recalculate the roles for every user: php update_roles.php refresh";
```

Comment:
this needs a couple carriage returns

Reply:
@taracampbell this also still applies, add carriage returns

Reply:
If you mean \n - @driusan told be to get rid of them